src/charm.py

```diff
@@ -48,6 +53,15 @@ def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR
         """
         ...
 
+    def callback(self, csr: bytes, relation_id: int) -> None:
```

Comment:
I decided to go with a different implementation, as there is a particular issue with this one. If the filter is responsible for saving the allowed CSRs, if it is enabled after some requirers already requested some certificates, things will not work properly.

I did some refactoring on my original PR to make the filter completely independent of what will be stored. The charm itself will be responsible for managing the peer relation, and transform the peer data into the right format for the filter.

Reply:
I thought it would be the expected behavior to not reserve anything if the reserve filter is disabled.